### PR TITLE
Implement find order command features for additional order attributes

### LINKED ISSUE #13: As a home baker with multiple orders, I want to track which order is for pick up or delivery
...so that I can give the correct orders to the delivery company and make enough space at home to store the cakes.

### PR DESCRIPTION
**Pull request by Geralddtan**
Lets implement find order for details, collectiontype and remarks and its corresponding predicates it uses for filtering
Lets also implement the necessary tests for find order - name, phone, details, collectiontype, remarks

Closes #13 & #33

### CODE REVIEW

File: src/main/java/seedu/address/model/order/CollectionTypeContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.order;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
```
Comment (onefangg) [inline src/main/java/seedu/address/model/order/CollectionTypeContainsKeywordsPredicate.java:6]: Update the comments here to "Order" and "CollectionType"?

File: src/main/java/seedu/address/model/order/DetailsContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package seedu.address.model.order;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
```
Comment (onefangg) [inline src/main/java/seedu/address/model/order/DetailsContainsKeywordsPredicate.java:9]: Update the comments here to "Order" and "Details"?

**Review by onefangg — APPROVED** [review]
Other than superficial changes, LGTM

File: src/main/java/seedu/address/model/order/CollectionTypeContainsKeywordsPredicate.java (hunk 1 above)
Comment (Geralddtan) [reply src/main/java/seedu/address/model/order/CollectionTypeContainsKeywordsPredicate.java:6]: Great spot!

**Pull request merged**